addons/hr_holidays/models/hr_leave.py

```diff
@@ -1658,7 +1656,13 @@ def _to_utc(self, date, hour, resource):
         holiday_tz = timezone(resource.tz or self.env.user.tz or 'UTC')
         return holiday_tz.localize(datetime.combine(date, hour)).astimezone(UTC).replace(tzinfo=None)
 
-    def _get_attendances(self, request_date_from, request_date_to, day_period=None):
+    def _get_hour_from_to(self, request_date_from, request_date_to, day_period=None):
+        """
+        Return the hour_from and hour_to for the given request dates, based on
+        the resource calendar.
+
+        If there are no attendances on the exact days of the request,
+        """
         self.ensure_one()
         domain = [
```

Comment:
Though it's not a ton of records, we could also restrict the search to the week_type dayofweek of both start and end date :thinking: